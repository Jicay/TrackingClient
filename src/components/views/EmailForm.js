import React, {Component} from "react";

export default class EmailForm extends Component {

    constructor(props) {
        super(props);
        this.state = {value: ''};

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(event) {
        this.setState({value: event.target.value});
    }

    handleSubmit(event) {
        event.preventDefault();
        this.props.history.push('/congrats');
    }

    render() {
        return (
            <form onSubmit={this.handleSubmit}>
                <div className="form-group">
                    <label htmlFor="inputMail">Email :</label>
                    <input type="email" id="inputMail" placeholder="Saisissez votre adresse e-mail" value={this.state.value} onChange={this.handleChange}/>
                </div>
                <button type="submit" className="btn btn-primary">Envoyer</button>
            </form>
        );
    }
}