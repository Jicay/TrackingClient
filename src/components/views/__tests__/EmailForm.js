import React from 'react';
import renderer from 'react-test-renderer';
import EmailForm from "../EmailForm";

it('renders correctly', () => {
    const tree = renderer
        .create(<EmailForm/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});