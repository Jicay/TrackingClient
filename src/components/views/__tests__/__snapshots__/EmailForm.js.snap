// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      htmlFor="inputMail"
    >
      Email :
    </label>
    <input
      id="inputMail"
      onChange={[Function]}
      placeholder="Saisissez votre adresse e-mail"
      type="email"
      value=""
    />
  </div>
  <button
    className="btn btn-primary"
    type="submit"
  >
    Envoyer
  </button>
</form>
`;
